src/PercentileStats.cpp

```diff
@@ -201,12 +202,19 @@ QRealList PercentileStats::percentileSums()
     {
         int percentile = qMax( 1, (int) ceil( i / percentileSize ) );
 
-        sums[ percentile ] += _data.at(i);
+        sums._ind[ percentile ] += _data.at(i);
     }
 
+    qreal runningTotal = 0;
+    for( int i=0; i < sums._ind.size(); i++)
+        sums._cum += (runningTotal += sums._ind.at(i));
```

Comment:
Please take this apart into two statements. Two `+=` assignments in one line leaves questions what exactly is added to the outer assignment. It doesn't matter if the rvalue expression needs to be evaluated twice; the loop runs only over a very small number of iterations.